src/definition.rs

```diff
@@ -113,11 +114,19 @@ impl TemplateDefinition {
                     continue;
                 }
                 Value::String(s) => {
+                    let variables = has_template_variables(&s, &vals);
+                    let default_value = match variables {
+                        Some(variables) => replace_with_previous_responses(
+                            variables, &vals, &s),
+                        None => s.clone(),
+                    };
```

Comment:
I think doing it that way is more complicated than it actually requires.
A default value will always be a valid Tera template as it's just a string so you can just check whether `{{` and `}}` are in the string and handle both cases:

1. no: just use that string a default, no need to do anything
2. yes: render that string as a one off template with the variables so far added to the Tera context.